src/lib/cache.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {existsSync} from 'fs';
+import {mkdir, readFile, stat, unlink, writeFile} from 'fs/promises';
+import {tmpdir} from 'os';
+import {join} from 'path';
+import {GitHubRepository, Issue, PullRequest} from './github';
+
+const cacheDirectory = join(tmpdir(), 'google-repo-cache');
+const cacheMaxAge = 60 * 60 * 1000; // 1 hour
+
+export type CachedData = {issues?: Issue[]; prs?: PullRequest[]};
+export type CacheType = 'prs' | 'issues';
+
+// Quick'n'dirty mutex implementation, because it spawns multiple workers
```

Comment:
WDYT about using a library like `async-mutex`?

Reply:
Good idea! Changed to `async-mutex` (it's pretty small by modern standards, just 200K in `node_modules`, so no reason not to use it).